lib/holepuncher.js

```diff
@@ -297,4 +305,55 @@ function coerceFirewall (fw) {
   return fw === FIREWALL.OPEN ? FIREWALL.CONSISTENT : fw
 }
 
+function localAddresses (socket) {
+  const addrs = []
+  const port = socket.address().port
+
+  for (const n of socket.udx.networkInterfaces()) {
+    if (n.family !== 4 || n.internal) continue
+
+    addrs.push({ host: n.host, port })
+  }
+
+  if (addrs.length === 0) {
+    addrs.push({ host: '127.0.0.1', port })
+  }
+
+  return addrs
+}
+
+function matchAddress (localAddresses, remoteLocalAddresses) {
+  if (remoteLocalAddresses.length === 0) return null
+
+  let best = { segment: 1, addr: null }
+
+  for (const localAddress of localAddresses) {
+    // => 192.168.122.238
+    const a = localAddress.host.split('.')
+
+    for (const remoteAddress of remoteLocalAddresses) {
+      // => 192.168.0.23
+      // => 192.168.122.1
+      const b = remoteAddress.host.split('.')
+
+      // Matches 192.*.*.*
+      if (a[0] === b[0]) {
+        if (best.segment === 1) best = { segment: 2, addr: remoteAddress }
+
+        // Matches 192.168.*.*
+        if (a[1] === b[1]) {
+          if (best.segment === 2) best = { segment: 3, addr: remoteAddress }
+
+          // Matches 192.168.122.*
+          if (a[2] === b[2]) {
+            if (best.segment === 3) best = { segment: 4, addr: remoteAddress }
+          }
+        }
+      }
+    }
+  }
```

Comment:
There should never be more addresses than would be feasible for a quadratic search to look through, no?

Reply:
Very true, before there was but code has changed